src/app/builders/components/BuilderRewards/hooks/useBuilderEstimatedRewards.ts

```diff
@@ -0,0 +1,182 @@
+import { useCycleContext } from '@/app/collective-rewards/metrics/context/CycleContext'
+import {
+  formatMetrics,
+  getBackerRewardPercentage,
+  useGetPerTokenRewards,
+} from '@/app/collective-rewards/rewards'
+import { useBuilderContext } from '@/app/collective-rewards/user'
+import { isBuilderRewardable, useHandleErrors } from '@/app/collective-rewards/utils'
+import { WeiPerEther } from '@/lib/constants'
+import { usePricesContext } from '@/shared/context/PricesContext'
+import { useReadBackersManager, useReadBuilderRegistry } from '@/shared/hooks/contracts'
+import { useReadGauge } from '@/shared/hooks/contracts/collective-rewards/useReadGauge'
+import { useReadRewardDistributor } from '@/shared/hooks/contracts'
+import { Address } from 'viem'
+import { Token } from '@/app/collective-rewards/rewards'
+import { useEffect, useMemo, useState } from 'react'
+
+interface UseBuilderEstimatedRewardsProps {
+  builder: Address
+  gauge: Address
+  tokens: {
+    rif: Token
+    rbtc: Token
+  }
+  currency?: string
+}
+
+interface TokenRewardData {
+  amount: string
+  fiatAmount: string
+  isLoading: boolean
+}
+
+interface EstimatedRewardsData {
+  rif: TokenRewardData
+  rbtc: TokenRewardData
+}
+
+export const useBuilderEstimatedRewards = ({
+  builder,
+  gauge,
+  tokens: { rif, rbtc },
+  currency = 'USD',
+}: UseBuilderEstimatedRewardsProps): EstimatedRewardsData => {
+  const { prices } = usePricesContext()
+  const { getBuilderByAddress } = useBuilderContext()
+
+  // Get per-token rewards
+  const { rif: rifRewards, rbtc: rbtcRewards } = useGetPerTokenRewards()
+
+  // State for RIF rewards
+  const [rifRewardsAmount, setRifRewardsAmount] = useState<bigint>(0n)
+  const [rifRewardsError, setRifRewardsError] = useState<Error | null>(null)
+  const [isRifRewardsLoading, setIsRifRewardsLoading] = useState(true)
+
+  // State for rBTC rewards
+  const [rbtcRewardsAmount, setRbtcRewardsAmount] = useState<bigint>(0n)
+  const [rbtcRewardsError, setRbtcRewardsError] = useState<Error | null>(null)
+  const [isRbtcRewardsLoading, setIsRbtcRewardsLoading] = useState(true)
+
+  // Update RIF rewards state
+  useEffect(() => {
+    if (rifRewards && rifRewards.data) {
+      setRifRewardsAmount(rifRewards.data ?? 0n)
+    }
+    if (rifRewards && rifRewards.error) {
+      setRifRewardsError(rifRewards.error)
+    }
+    if (rifRewards) {
```

Comment:
```suggestion
    if (rifRewards && rifRewards.isLoading) {
```

Reply:
I've checked again this and I'm no longer sure about the suggested changes. I think we should check `isLoading` against the previous value of `isRifRewardsLoading`